homeassistant/components/smartthings/__init__.py

```diff
@@ -128,14 +169,19 @@
 class DeviceBroker:
     """Manages an individual SmartThings config entry."""
 
-    def __init__(self, hass: HomeAssistantType, devices: Iterable,
-                 installed_app_id: str):
+    def __init__(self, hass: HomeAssistantType, entry: ConfigEntry,
+                 token, smart_app, devices: Iterable):
         """Create a new instance of the DeviceBroker."""
         self._hass = hass
-        self._installed_app_id = installed_app_id
-        self.assignments = self._assign_capabilities(devices)
+        self._entry = entry
+        self._installed_app_id = entry.data[CONF_INSTALLED_APP_ID]
+        self._smart_app = smart_app
+        self._token = token
+        self._event_disconnect = None
+        self._regenerate_token_remove = None
+        self._assignments = self._assign_capabilities(devices)
         self.devices = {device.device_id: device for device in devices}
-        self.event_handler_disconnect = None
+        self._connect()
```

Comment:
We try to avoid side effects in init method.

Reply:
Removed and made public